client/app/components/EditableDocumentField.jsx

```
@@ -0,0 +1,102 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Button from './Button';
+
+export default class EditableDocumentField extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      editing: false
+    };
+  }
+
+  inputRef = (node) => this.input = node;
+
+  startEditing = () => this.setState({ editing: true });
+  stopEditing = () => this.setState({ editing: false });
+
+  onSave = () => {
+    this.props.onSave(this.input.value);
+    this.stopEditing();
+  }
+  onCancel = () => {
+    this.props.onCancel();
+    this.stopEditing();
+  }
+  onChange = (event) => this.props.onChange(event.target.value);
+
+  componentDidUpdate = () => {
+    if (this.props.errorMessage && !this.state.editing) {
```

We probably just want to do this if the errorMessage prop has changed. Not positive though.

---

Sure, I guess it's a matter of which component should handle processing the error on behalf of `EditableField` (it vs. its parent).

At the moment, `SidebarDocumentInformation` passes `this.props.error.visible ? this.props.error.message : null`, so checking `!!this.props.errorMessage` should be sufficient. Hiding the error message happens within `PdfActions.changePendingDocDescription`, which is passed as `EditableField.onChange`.

It seemed easier to have the parent pass an error if needed than to have the child be aware of some part of the store, but I guess with EditableField having only one use-case for now, it could go either way